src/referencePicker/RawLinkInput.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+	<div class="raw-link">
+		<div class="input-wrapper">
+			<NcTextField
+				ref="url-input"
+				:value.sync="inputValue"
+				:show-trailing-button="inputValue !== ''"
+				:label="inputPlaceholder"
+				@trailing-button-click="onClear"
+				@update:value="onInput"
+				@keyup.native.enter="onSubmit">
+				<template #trailing-button-icon>
+					<CloseIcon :size="16" />
+				</template>
+				<NcLoadingIcon v-if="loading" :size="16" />
+				<LinkVariantIcon v-else :size="16" />
+			</NcTextField>
+		</div>
+		<ReferenceWidget v-if="reference !== null"
+			class="reference-widget"
+			:reference="reference" />
+		<NcEmptyContent v-else
+			class="raw-link--empty-content">
+			<template #icon>
+				<img v-if="provider.icon_url"
+					 class="provider-icon"
+					 :src="provider.icon_url">
+				<LinkVariantIcon v-else />
+			</template>
+		</NcEmptyContent>
+	</div>
+</template>
+
+<script>
+import LinkVariantIcon from 'vue-material-design-icons/LinkVariant.vue'
+
+import NcEmptyContent from '@nextcloud/vue/dist/Components/NcEmptyContent.js'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+
+import ReferenceWidget from '../ReferenceWidget.vue'
+import CustomPickerElement from './CustomPickerElement.vue'
+import { isUrl, delay } from './utils.js'
+import axios from '@nextcloud/axios'
+import { generateOcsUrl } from '@nextcloud/router'
+
+export default {
+	name: 'RawLinkInput',
+	components: {
+		ReferenceWidget,
+		LinkVariantIcon,
+		CustomPickerElement,
+		NcEmptyContent,
+		NcTextField,
+		NcLoadingIcon,
+	},
+	props: {
+		/**
+		 * The reference provider
+		 */
+		provider: {
+			type: Object,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			inputValue: '',
+			loading: false,
+			reference: null,
+			abortController: null,
+			// TODO translate?
```

Comment:
Let's have this as a follow up task so we can get a first release out.

Reply:
Ok and let's keep the comments if you don't mind.